posthog/models/file_system/user_product_list.py

```diff
@@ -0,0 +1,54 @@
+from django.db import models
+from django.db.models.expressions import F
+
+from posthog.models.team import Team
+from posthog.models.user import User
+from posthog.models.utils import UpdatedMetaFields, UUIDModel, uuid7
+
+
+class UserProductList(UUIDModel, UpdatedMetaFields):
+    """
+    Stores a user's custom list of products they care about.
+    Products are identified by their path from the static products list.
+    """
+
+    id = models.UUIDField(primary_key=True, default=uuid7, editable=False)
+    team = models.ForeignKey(Team, on_delete=models.CASCADE)
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    product_path = models.CharField(max_length=200)
+
+    # Not using `CreatedMetaFields` because of the clashing `user` reference with `created_by`
+    created_at = models.DateTimeField(auto_now_add=True)
+
+    class Reason(models.TextChoices):
+        # User used the product enough to warrant it being in the sidebar
+        USAGE = "usage", "Usage"
+
+        # We launch a new product and want to foster adoption
+        NEW_PRODUCT = "new_product", "New Product"
```

Comment:
So this means we need to create a new row for every user per product as a batch operation when we want to promote a new product... plus also auto-generate it for new users? We might need to turn it around, so people can opt out of promoted products instead?

Reply:
New users won't have all products in the sidebar, they'll only get the ones they request in the onboarding. We'll eventually suggest new products to then following some heuristics - that doesn't exist yet - which is just the default flow for users.

The rest of your assumption is correct, yes :)